README.md

```diff
@@ -21,6 +18,65 @@ python -m pip install .
+
+    ```bash
+    make dvc-init
+    make dvc-autostage
+    ```
+
+4. The previous command will have created a `.dvc` directory and a `.dvcignore` file. You will also see a few new files in the Git staging area. Commit these changes to Git with a message like "Set up DVC". Now you are ready to start using `dm-era5` with DVC.
+
+### Download ERA5 data
+
+Modify the `era5-download` section of the `config.ini` file to specify the desired download parameters. Then run:
+
+```bash
+python -m dmd_era5.era5_download.era5_download
+```
+
+Note that depending on the size of the download, this may be a memory-intensive process and you might need to run it on a HPC. The downloaded data will be saved in the `data/era5_download` directory as a NetCDF file, using the time range and time delta specified in `config.ini` as the file name (e.g. `2019-01-01T00_2019-01-02T00_1h.nc` for a time range from 2019-01-01 00:00 to 2019-01-02 00:00 with a time delta of 1 hour).
+
+If you followed the DVC setup instructions above, the downloaded data will be automatically tracked by DVC, and you will see three new files appearing in your Git staging area. For example, if you downloaded the file `2019-01-01T00_2019-01-02T00_1h.nc`, you will see the following files in the Git staging area:
+
+- `data/era5_download/.gitignore`, automatically created by DVC to tell Git to ignore the downloaded data file.
+- `data/era5_download/2019-01-01T00_2019-01-02T00_1h.nc.dvc`, automatically created by DVC to track the downloaded data version with a unique hash.
+- `data/era5_download/2019-01-01T00_2019-01-02T00_1h.nc.yaml`, which is a human-readable log of the different versions of the downloaded data, showing the metadata of each version. You can open this file to see the different versions of `2019-01-01T00_2019-01-02T00_1h.nc` that you have downloaded.
+
+You should now commit these files to Git with a message like "First download of 2019-01-01T00_2019-01-02T00_1h.nc". If you have set up a remote storage location for your data, you can push the data to the remote storage location by running:
+
+```bash
+dvc push
+```
+
+Once you have pushed to the remote, it is safe to delete the downloaded NetCDF file from `data/era5_download` if you wish, and you can even delete the DVC cache (which stores the data versions locally) by running `rm -rf .dvc/cache`. Note that deleting the data stored locally is an optional step that might be useful if you are running out of disk space.
+
+Next time you want to download the same data or a subset of it (i.e. a subset of variables or pressure levels) over the same time range and time delta and from the same source, you can simply run `python -m dmd_era5.era5_download.era5_download` again. DVC will automatically detect that you have already downloaded the data and will not download it again; it will simply retrieve the right version of the data from the DVC repository.
```

Comment:
A note here after discussing with @dsj976, adding capability to create folder if it doesn't exist. Else users like me who are too eager to delete folders would run into this problem (deleted era5_download rather than the files inside it)

```
2024-11-27 18:44:17,222 - ERA5Download - INFO - Attempting to retrieve ERA5 slice from DVC...
Attempting to retrieve ERA5 slice from DVC...
2024-11-27 18:44:17,223 - ERA5Download - INFO - Could not retrieve ERA5 slice from DVC: DVC file or log file does not exist.
Could not retrieve ERA5 slice from DVC: DVC file or log file does not exist.
2024-11-27 18:44:17,223 - ERA5Download - INFO - Loading ERA5 Dataset...
Loading ERA5 Dataset...
2024-11-27 18:44:44,001 - ERA5Download - INFO - ERA5 loaded.
ERA5 loaded.
2024-11-27 18:44:44,001 - ERA5Download - INFO - Slicing ERA5 Dataset...
Slicing ERA5 Dataset...
2024-11-27 18:44:44,033 - ERA5Processing - INFO - Dataset slicing completed successfully using 2019-01-01 00:00:00to 2019-01-05 00:00:00 and levels [1000]
Dataset slicing completed successfully using 2019-01-01 00:00:00to 2019-01-05 00:00:00 and levels [1000]
2024-11-27 18:44:44,034 - ERA5Download - INFO - Size of ERA5 Dataset: 403.0 MB
Size of ERA5 Dataset: 403.0 MB
2024-11-27 18:44:44,034 - ERA5Download - INFO - Saving ERA5 Dataset to data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc...
Saving ERA5 Dataset to data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc...
2024-11-27 18:44:44,054 - ERA5Download - ERROR - 
        Error downloading ERA5 Dataset: [Errno 13] Permission denied: '/bask/projects/v/vjgo8416-climate/users/sgft6713/DMD-ERA5/data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc'
        

        Error downloading ERA5 Dataset: [Errno 13] Permission denied: '/bask/projects/v/vjgo8416-climate/users/sgft6713/DMD-ERA5/data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc'
        
2024-11-27 18:44:44,054 - ERA5Download - ERROR - Configuration error: 
        Error downloading ERA5 Dataset: [Errno 13] Permission denied: '/bask/projects/v/vjgo8416-climate/users/sgft6713/DMD-ERA5/data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc'
        
Configuration error: 
        Error downloading ERA5 Dataset: [Errno 13] Permission denied: '/bask/projects/v/vjgo8416-climate/users/sgft6713/DMD-ERA5/data/era5_download/2019-01-01T00_2019-01-05T00_1h.nc'

```

Reply:
Other than that, after adding the folder era5_download back into the data folder, it runs perfectly :). I was also able to run pytest without problems.